### ISSUE TITLE
anchor after test command in a namespace produces duplicate tests

### ISSUE THREAD

**Issue by calumblair** [doxygen/doxygen#6589]
Minimal working example:
main.h:
```
/** foo
*/
namespace foo{
/** 
\anchor anchor_1
\test f() should exhibit behaviour A
\anchor anchor_2
\test f() should exhibit behaviour B
*/
void f(int);
};
```

In the generated test list, this produces:
```
Member foo::f (int)

    f() should exhibit behaviour A

    f() should exhibit behaviour B

    f() should exhibit behaviour B

```
i.e. the second `\test` tag is incorrectly duplicated.
The bug goes away if I move anchor_2 above both `\test` commands:
```
\anchor anchor_1
\anchor anchor_2
\test f() should exhibit behaviour A
\test f() should exhibit behaviour B
```
But I'm using an alias to combine one `\anchor` and one `\test` into a single command, so this is not possible.
This behaviour is observed on doxygen-master and doxygen 1.8.5 using the generated Doxyfile config.

**Comment by albert-github** [comment]
Can you please attach a, small, self contained example (source+config file in a tar or zip) that allows us to reproduce the problem?

**Comment by calumblair** [comment]
[anchor_namespace_doxygen_bug.zip](https://github.com/doxygen/doxygen/files/2538352/anchor_namespace_doxygen_bug.zip)

See attached. Run doxygen then look at the Related Files->Test List page.

**Comment by albert-github** [comment]
Is there a special reason why you add an extra anchor?
(I can reproduce the problem, but I'm just curious about it)

**Comment by calumblair** [comment]
This allows me to define multiple requirements in a source file and link it to unit tests. In the test list, I'll then have links to the source which implement the requirements, and the tests which verify them.
So I can see that all requirements are tested.
But because of this bug, I get duplicated tests in the test list.
[requirements_doxygen_bug.zip](https://github.com/doxygen/doxygen/files/2538473/requirements_doxygen_bug.zip)

In Doxyfile:
```
ALIASES += requirement{2}="\anchor src_\1 \test \b \1 : \2 | \ref src_\1 \"SRC\" | \ref tst_\1 \"TEST\" \n"
ALIASES += verifies{1}="\anchor tst_\1 \ref src_\1 \"SRC\"\n"
```

In a source file:
```
/** foo
*/
namespace foo{
/**
\requirement{g_returns_0_if_valid, g must return 0 if input is >0}
\requirement{g_returns_1_if_invalid, g must return 1 if input is <0}
*/
int g(int);
};
```
In a test file:
```
/**
call g() and check it returns 0 if given valid input
\verifies{g_returns_0_if_valid}
*/
check_g_returns_zero(1);

/**
call g() and check it returns 1 for invalid input
\verifies{g_returns_1_if_invalid}
*/
check_g_returns_one(-1);
```

This will produce in the test list:
```
Member foo::g (int)

    g_returns_0_if_valid : g must return 0 if input is >0 | SRC | TEST

    g_returns_1_if_invalid : g must return 1 if input is <0 | SRC | TEST

    g_returns_1_if_invalid : g must return 1 if input is <0 | SRC | TEST
```
In the HTML, SRC and TEST are links to the relevant anchors. Here, the second `g_returns_1_if_invalid`  is due to this bug.

**Comment by albert-github** [comment]
I understand it is for extra cross referencing. I think this should work.

**Comment by albert-github** [comment]
Code has been integrated in master on github.

**Comment by calumblair** [comment]
This is working now. Thanks!

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.15.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
